### PR TITLE
feat(linter/plugins): implement `SourceCode#getLastToken()`

### PR DESCRIPTION
**Pull request by lilnasy**
- Part of https://github.com/oxc-project/oxc/issues/14829#issuecomment-3512679496.
- Follow up to #15861.

### CODE REVIEW

File: apps/oxlint/src-js/plugins/tokens.ts [hunk 1]
```diff
@@ -385,11 +385,96 @@ export function getFirstTokens(node: Node, countOptions?: CountOptions | number
...
+      hi = mid;
+    }
+  }
+
+  // TODO: this early return feels iffy
+  if (lastTokenIndex === nodeTokensLength) return null;
+
+  if (typeof filter !== 'function') {
+    if (typeof skip !== 'number') {
+      return nodeTokens[lastTokenIndex] ?? null;
```
[the remark this answers is not in the document]
Comment (overlookmotel) [reply apps/oxlint/src-js/plugins/tokens.ts:565]: I don't know if this comment is reasonable or not? @lilnasy Please can you evaluate his claim and post response here?

[the remark this answers is not in the document]
Comment (lilnasy) [reply apps/oxlint/src-js/plugins/tokens.ts:565]: Yes, it is valid. Thanks for the ping. I meant to implement this, but it slipped through the cracks.

[the remark this answers is not in the document]
Comment (lilnasy) [reply apps/oxlint/src-js/plugins/tokens.ts:565]: I spoke too soon. It is only valid for manually written ranges, not for last tokens of actual, correctly parsed nodes, which logically must have at least one full token. Node ranges cannot start or stop in the middle of token ranges.

[the remark this answers is not in the document]
Comment (overlookmotel) [reply apps/oxlint/src-js/plugins/tokens.ts:565]: I think we can assume that the `node` provided is a valid AST node object.

The only edge case we need to handle is an empty file, where `Program` is a valid node, but can contain 0 tokens (I only realized this odd case from one of the test cases in one of your previous PRs). But you handle that with `if (lastTokenIndex === nodeTokensLength) return null;`, so that's covered. All good.
